File: IQ/Classes/clsROKAttribute.vb
Option Explicit On
Option Strict On

Imports dataAccess
Imports System.IO

' Represents a collection of OS ROK attributes

Public Class clsROKAttribute

    Public ID As Integer
    Public OsCode As String
    Public Code As String
    Public Translation As clsTranslation

    Public Sub New(ByVal ID As Integer, ByVal OsCode As String, ByVal Code As String, ByVal Translation As clsTranslation)

        Me.ID = ID
        Me.OsCode = OsCode
        Me.Code = Code
        Me.Translation = Translation

    End Sub

    Public Sub update()

        Dim sql$ = "UPDATE ROKAttributes set fk_translation_key_name=" & Me.Translation.Key & " WHERE id = " & Me.ID
        da.DBExecutesql(sql)

    End Sub

End Class

File: IQ/Classes/clsROKAttribute.cs
using dataAccess;
using System.IO;



// Represents a collection of OS ROK attributes

public class clsROKAttribute
{

    public int ID;
    public string OsCode;
    public string Code;
    public clsTranslation Translation;

    public clsROKAttribute(int ID, string OsCode, string Code, clsTranslation Translation)
    {

        this.ID = ID;
        this.OsCode = OsCode;
        this.Code = Code;
        this.Translation = Translation;

    }

    public void update()
    {

        string sql = "UPDATE ROKAttributes set fk_translation_key_name=" + this.Translation.Key + " WHERE id = " + System.Convert.ToString(this.ID);
        da.DBExecutesql(sql);

    }

}